src/test/scala/zio/prelude/fx/ZPureSpec.scala

```diff
@@ -187,7 +188,7 @@ object ZPureSpec extends DefaultRunnableSpec {
           },
           testM("orElseSucceed (Success case)") {
             check(genInt, genInt, genInt) { (s1, v, v1) =>
-              val (_, a) = ZPure.succeed(v).orElseSucceed(v1).run(s1)
+              val (_, a) = ZPure.succeed(v).orElseSucceed(v1)(CanFail).run(s1)
```

Comment:
I'm not sure we should specify this explicitly. On the opposite, shouldn't this test verify, that when actual users will use this in their code, they won't have to specify it either?

Reply:
The user shouldn't ever have to specify it. For testing in some cases we may need ignore the `CanFail` warnings. For example here I think the intent is to test that when you call `orElseSucceed` on a successful value you just get the original success back, which makes sense but would normally cause an error.

A slightly nicer way to do this is to do the following within the scope of your test:

```scala
implicit val canFail = CanFail
```

This will silence the warnings for that scope.

Reply:
@adamgfraser Ok, I'll switch to the implicit declaration